### DISCUSSION TITLE
Dropping SQLite support

### DISCUSSION THREAD

**Discussion by TheAxelander** [TheAxelander/OpenBudgeteer#242] [Announcements]
As mentioned in #239 I'm making now the decision of dropping support for SQLite database entirely. Over the time new issues were popping up again and again and I'm tired of fixing these and coding exception routes for SQLite-only. My goal is to focus on a stable and reliable database connection (which will be for now MariaDB), instead of supporting multiple database provider.

There will be some cleanup activities required in the coding, plus I need to rewrite my Unit Testing which is currently based on multiple isolated SQLite databases.

For now I will remove any reference to SQLite in the documentation for version `1.8.x`, the full cleanup will then happen in `pre-release` and finally released with the next version.

**Comment by Anu6is** [comment]
Would it be possible to leave the existing support available in the core code but remove the docker support? 
I find SQLite especially useful for spinning up a temp db when developing. It also mean there's no real database setup required when deploying.

**Comment by TheAxelander** [comment]
I will think about it as I have not yet decided what kind of cleanup I will do at the end, but definitely I will no longer update the SQLite coding in terms of new features. So on the long term perspective the remains will not support all features and/or simply break due to incompatibility
